common/thorhelper/thorsoapcall.cpp

```diff
@@ -170,6 +170,24 @@ class Url : public CInterface, implements IInterface
         return;
     }
 
+    void cleanPath()
+    {
+        size32_t len = path.length();
+        if (len <= 1)
+            return;
+        size32_t gap = 0;
+        char* buf = path.detach();
+        //Remove all duplicate slashes
+        for (size32_t i = 1; i < len; i++)
+        {
+            if (buf[i] == '/' && buf[i-1-gap] == '/')
```

Comment:
minor: [i-1-gap] is always the 1st '/' in a sequence, so you could code this more efficiently as a while loop.

Personally I'd code this routine something like this:
```
void cleanPath(const char *src, StringBuffer &output)
{
    const char *srcPtr = src;
    char *dst = output.reserveTruncate(strlen(src));
    char *dstPtr = dst;
    while (*srcPtr)
    {
        *dstPtr++ = *srcPtr;
        if ('/' == *srcPtr++)
        {
            while ('/' == *srcPtr)
                ++srcPtr;
        }
    }
    *dstPtr = '\0';
    output.setLength(dstPtr-dst);
}

```

You could pass a single StringBuffer as both the input and output, and detach() instead, but given the input in the Uri ctor isn't originally a StringBuffer (i.e. 'p' was just appended before calling cleanPath)...